muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/notes/NotesRESTService.java

```diff
@@ -0,0 +1,171 @@
+  
+  @Inject
+  private SessionController sessionController;
+  
+  @Inject
+  private NotesController notesController;
+  
+
+  // mApi() call (mApi().notes.note.create(noteRestModel)
+  //
+  // noteRestModel: = {
+  //  title: String, 
+  //  description: String, 
+  //  type: emun MANUAL
+  //  priority: enum LOW/NORMAL/HIGH, 
+  //  pinned: Boolean, 
+  //  owner: userIdentifier,
+  //  creator: loggedUser
+  //  };
+  @POST
+  @Path("/note")
+  @RESTPermit (handling = Handling.INLINE, requireLoggedIn = true)
+  public Response createNote(NoteRestModel note) {
+    
+    if (!sessionController.isLoggedIn()) {
+      return Response.status(Status.UNAUTHORIZED).build();
+    }
+    
+    if (!sessionController.hasEnvironmentPermission(NotesPermissions.NOTES_CREATE)) {
+      return Response.status(Status.FORBIDDEN).build();
+    }
+    
+    try {
+      SchoolDataIdentifier loggedUser = sessionController.getLoggedUser();
+      
+      Note newNote = notesController.createNote(note.getTitle(), note.getDescription(), note.getType(), note.getPriority(), note.getPinned(), note.getOwner(), loggedUser.toString(), new Date());
+      return Response.ok(restModel(newNote)).build();
+    }
+    catch (Exception e) {
+      e.printStackTrace();
+      logger.log(Level.SEVERE, "Couldn't create new note.", e);
+      return Response.status(Status.INTERNAL_SERVER_ERROR).build();
+    } 
+  }
+  
+  //mApi() call (mApi().notes.note.update(noteId, noteRestModel)
+  // Editable fields are title, description, priority and pinned)
+  @PUT
+  @Path ("/note/{NOTEID}")
+  @RESTPermit(handling = Handling.INLINE)
+  public Response updateNote(@PathParam ("NOTEID") Long noteId, NoteRestModel restModel) {
+    
+    
+    if (!sessionController.isLoggedIn()) {
+      return Response.status(Status.UNAUTHORIZED).build();
+    }
+    
+    Note note = notesController.findNoteById(noteId);
+    
+    if (note == null) {
+      return Response.status(Status.NOT_FOUND).build();
+    }
+    
+    SchoolDataIdentifier ownerIdentifier = SchoolDataIdentifier.fromId(note.getOwner());
+    
+    //permissions 
+    
+    if (!sessionController.hasEnvironmentPermission(NotesPermissions.NOTES_UPDATE)) {
+      if (!Objects.equals(sessionController.getLoggedUser(), ownerIdentifier))
+        return Response.status(Status.FORBIDDEN).build();
+    }
+    
+    notesController.updateNote(note, restModel.getTitle(), restModel.getDescription(), restModel.getPriority(), restModel.getPinned());
+  
+    return Response.noContent().build();
+  }
+  
+  
+  
+  private NoteRestModel restModel(Note note) {
+    NoteRestModel restModel = new NoteRestModel(note.getId(), note.getTitle(), note.getDescription(), note.getType(), note.getPriority(), note.getPinned(), note.getOwner(), note.getCreator(), note.getCreated(), note.getLastModifier(), note.getLastModified(), note.getArchived());
+
+    return restModel;
+  }
+  
+  //mApi() call (mApi().notes.read(owner))
+  @GET
+  @Path("/{OWNER}")
```

Comment:
`@Path("/owner/{OWNER}")` might be safer in the long run, just in case we ever list notes in some other fashion.